crypto/cms/cms_env.c

```diff
@@ -242,6 +242,42 @@ CMS_ContentInfo *CMS_EnvelopedData_create(const EVP_CIPHER *cipher)
     return CMS_EnvelopedData_create_ex(cipher, NULL, NULL);
 }
 
+BIO *CMS_EnvelopedData_decrypt(CMS_EnvelopedData *env, BIO *detached_data,
+                               EVP_PKEY *pkey, X509 *cert,
+                               ASN1_OCTET_STRING *secret, unsigned int flags)
+{
+    CMS_ContentInfo *ci;
+    BIO *bio = NULL;
+    int res = 0;
+
+    if (env == NULL) {
+        ERR_raise(ERR_LIB_CMS, ERR_R_PASSED_NULL_PARAMETER);
+        return NULL;
+    }
+
+    if ((ci = CMS_ContentInfo_new()) == NULL
```

Comment:
Is this supposed to be using a libctx?

Reply:
> Is this supposed to be using a libctx?

Good question - I don't know.
It would be no problem to add `libctx` and `propq` parameters to `CMS_EnvelopedData_decrypt()`.
On the other hand, `CMS_decrypt()` and `CMS_EncryptedData_decrypt()` do not have them.
Would it be beneficial to  add them just for calling ` CMS_ContentInfo_new_ex()` in this line?

Reply:
Should the libctx and propq be obtained from the env parameter instead?

Reply:
Please add the libctx and propq parameters. New API calls should be libctx aware so if the libctx cannot be inferred from some other parameter, there definitely must be a way to set it (and the propq).

Reply:
Oh, I just noticed that I missed your response from 19 days ago.
Done adding the libctx and propq parameters.